src/algebra/module/linear_map.lean

```diff
@@ -503,32 +562,62 @@ variables (M R)
 /-- The identity map is a linear equivalence. -/
 @[refl]
 def refl [module R M] : M ≃ₗ[R] M := { .. linear_map.id, .. equiv.refl M }
+
 end
 
 @[simp] lemma refl_apply [module R M] (x : M) : refl R M x = x := rfl
 
+include module_M module_S_M₂ re₁ re₂
 /-- Linear equivalences are symmetric. -/
 @[symm]
-def symm : M₂ ≃ₗ[R] M :=
-{ .. e.to_linear_map.inverse e.inv_fun e.left_inv e.right_inv,
+def symm (e : M ≃ₛₗ[σ] M₂) : M₂ ≃ₛₗ[σ'] M :=
+{ to_fun := e.to_linear_map.inverse e.inv_fun e.left_inv e.right_inv,
+  inv_fun := e.to_equiv.symm.inv_fun,
+  map_smul' := λ r x, by simp,
+  .. e.to_linear_map.inverse e.inv_fun e.left_inv e.right_inv,
   .. e.to_equiv.symm }
+omit module_M module_S_M₂ re₁ re₂
 
 /-- See Note [custom simps projection] -/
-def simps.symm_apply [module R M] [module R M₂] (e : M ≃ₗ[R] M₂) : M₂ → M := e.symm
+def simps.symm_apply {R : Type*} {S : Type*} [semiring R] [semiring S] {σ : R →+* S}
+  {σ' : S →+* R} [ring_hom_inv_pair σ σ'] [ring_hom_inv_pair σ' σ]
+  {M : Type*} {M₂ : Type*} [add_comm_monoid M] [add_comm_monoid M₂] [module R M] [module S M₂]
+  (e : M ≃ₛₗ[σ] M₂) : M₂ → M := e.symm
 
 initialize_simps_projections linear_equiv (to_fun → apply, inv_fun → symm_apply)
 
+include σ'
 @[simp] lemma inv_fun_eq_symm : e.inv_fun = e.symm := rfl
-
-variables {module_M₃ : module R M₃} (e₁ : M ≃ₗ[R] M₂) (e₂ : M₂ ≃ₗ[R] M₃)
-
-/-- Linear equivalences are transitive. -/
-@[trans]
-def trans : M ≃ₗ[R] M₃ :=
-{ .. e₂.to_linear_map.comp e₁.to_linear_map,
-  .. e₁.to_equiv.trans e₂.to_equiv }
-
-infixl ` ≪≫ₗ `:80 := linear_equiv.trans
+omit σ'
+
+variables {module_M₁ : module R₁ M₁} {module_M₂ : module R₂ M₂} {module_M₃ : module R₃ M₃}
+variables {module_N₁ : module R₁ N₁} {module_N₂ : module R₁ N₂}
+variables {σ₁₂ : R₁ →+* R₂} {σ₂₃ : R₂ →+* R₃} {σ₁₃ : R₁ →+* R₃}
+variables {σ₂₁ : R₂ →+* R₁} {σ₃₂ : R₃ →+* R₂} {σ₃₁ : R₃ →+* R₁}
+variables [ring_hom_comp_triple σ₁₂ σ₂₃ σ₁₃]
+variables [ring_hom_comp_triple σ₃₂ σ₂₁ σ₃₁]
+variables {re₁₂ : ring_hom_inv_pair σ₁₂ σ₂₁} {re₂₃ : ring_hom_inv_pair σ₂₃ σ₃₂}
+variables [ring_hom_inv_pair σ₁₃ σ₃₁] {re₂₁ : ring_hom_inv_pair σ₂₁ σ₁₂}
+variables {re₃₂ : ring_hom_inv_pair σ₃₂ σ₂₃} [ring_hom_inv_pair σ₃₁ σ₁₃]
+variables (e₁₂ : M₁ ≃ₛₗ[σ₁₂] M₂) (e₂₃ : M₂ ≃ₛₗ[σ₂₃] M₃)
+
+include σ₃₁
+/-- Linear equivalences are transitive. The linter thinks the `ring_hom_comp_triple` argument
```

Comment:
Can the comment on the linter be moved to a regular comment? I don't think it belongs to the true docstring.